Sources/Shared/Models/ConnectedApp.swift

```diff
@@ -0,0 +1,237 @@
+/// - linkedin: Represents a connection to LinkedIn.
+/// - twitter: Represents a connection to Twitter.
+/// - youtube: Represents a connection to YouTube.
+@objc public enum ConnectedAppType: Int {
+    case facebook
+    case linkedin
+    case twitter
+    case youtube
+
+    public var stringValue: String {
+        switch self {
+        case .facebook:
+            return String.facebook
+        case .linkedin:
+            return String.linkedin
+        case .twitter:
+            return String.twitter
+        case .youtube:
+            return String.youtube
+        }
+    }
+}
+
+///  An object that encapsulates the scopes necessary for interacting with features like publishing to social platforms
+///  or simulcasting a live stream.
+@objcMembers
+public class ConnectedAppScopes: VIMModelObject {
+
+    /// All scopes required for publishing to a specific social media platform.
+    public var publishToSocial: [String]?
+
+    /// All scopes required for simulcasting to a specific social media platform.
+    public var simulcast: [String]?
+
+    // MARK: - Overrides
+
+    public override func getObjectMapping() -> Any? {
+        return [
+            String.Key.publishToSocial: String.Value.publishToSocial
+        ]
+    }
+}
+
+/// A category that can be sent when publishing to a social media platform.
+@objcMembers
+public class PublishOptionItem: VIMModelObject {
+
+    /// The ID of the publish item.
+    public var identifier: String?
+
+    /// The name or display name of the publich item, i.e.: "art", "family", "vacation" etc.
+    public var name: String?
+
+    // MARK: - Overrides
+
+    public override func getObjectMapping() -> Any? {
+        return [
+            String.Key.identifier: String.Value.identifier,
+        ]
+    }
+}
+
+/// A `ConnectedApp` represents a connection to a social media platform. Some activities, like simultaneously live
+/// stream to multiple destinations, or publishing across platforms, require requesting specific scopes. The scopes
+/// required will always be returned in the `neededScopes` array.
+/// - Note: Some properties are specific to a particular platform. These cases have been noted in the documentation
+///         where relevant.
+@objc public class ConnectedApp: VIMModelObject {
+
+    /// The date when the user established this connected app, as a String.
+    /// - Note: This property is available to provide interoperability with Objective-C codebases.
+    ///         Using `addDate` is preferred.
+    @objc public var addDateString: String?
+
+    /// The date when the user established this connected app.
+    @objc public var addDate: Date?
+
+    /// Facebook only. A value of 1 maps to expired data access. A value of 0 maps to having data access.
+    /// - Note: This property is available to provide interoperability with Objective-C codebases.
+    ///         Using `isDataAccessExpired` is preferred.
+    @objc public var dataAccessIsExpired: NSNumber?
```

Comment:
does this need to be public given the computed boolean below?

Reply:
Good call, it does not.